Externals/SDL/CMakeLists.txt

```
@@ -1,34 +1,7 @@
-option(SDL2_DISABLE_SDL2MAIN "" ON)
-option(SDL2_DISABLE_INSTALL "" ON)
-option(SDL2_DISABLE_UNINSTALL "" ON)
-option(SDL_SHARED "Build a shared version of the library" OFF)
-option(SDL_SHARED_ENABLED_BY_DEFAULT "" OFF)
-option(SDL_STATIC "Build a static version of the library" ON)
-option(SDL_STATIC_ENABLED_BY_DEFAULT "" ON)
-option(SDL_TEST "Build the SDL2_test library" OFF)
-option(SDL_TEST_ENABLED_BY_DEFAULT "" OFF)
-
-# SDL fails to clean up old headers after version upgrades, so do that manually
-set(EXPECTED_SDL_REVISION "SDL-release-2.30.9-0")
-if (EXISTS "${CMAKE_CURRENT_BINARY_DIR}/SDL/include/SDL2/SDL_revision.h")
-  file(READ "${CMAKE_CURRENT_BINARY_DIR}/SDL/include/SDL2/SDL_revision.h" ACTUAL_SDL_REVISION)
-  if (NOT "${ACTUAL_SDL_REVISION}" MATCHES "${EXPECTED_SDL_REVISION}")
-    message(STATUS "Found unexpected SDL2/SDL_revision.h, removing generated includes.")
-    file(REMOVE_RECURSE "${CMAKE_CURRENT_BINARY_DIR}/SDL/include/")
-  endif()
-endif()
-if (EXISTS "${CMAKE_CURRENT_BINARY_DIR}/SDL/include/SDL_revision.h")
-  file(READ "${CMAKE_CURRENT_BINARY_DIR}/SDL/include/SDL_revision.h" ACTUAL_SDL_REVISION)
-  if (NOT "${ACTUAL_SDL_REVISION}" MATCHES "${EXPECTED_SDL_REVISION}")
-    message(STATUS "Found unexpected SDL_revision.h, removing generated includes.")
-    file(REMOVE_RECURSE "${CMAKE_CURRENT_BINARY_DIR}/SDL/include/")
-  endif()
-endif()
-
 add_subdirectory(SDL)
```

I think we still want `SDL_STATIC` to be `ON` and `SDL_SHARED` to be OFF, since we don't want ship SDL separately. [By default, SDL3 is shared](https://wiki.libsdl.org/SDL3/README-cmake#shared-or-static).

[Here is a list of the CMake options](https://wiki.libsdl.org/SDL3/README-cmake#sdl-specific-cmake-options)

```suggestion
set(SDL_SHARED OFF)
set(SDL_STATIC ON)
set(SDL_TEST_LIBRARY OFF)
set(SDL_TESTS OFF)
set(SDL_DISABLE_INSTALL ON)
set(SDL_DISABLE_INSTALL_DOCS ON)
set(SDL_INSTALL_TESTS OFF)

add_subdirectory(SDL)
```